src/resty/gcp/init.lua

```diff
@@ -108,20 +97,41 @@ local BuildMethods = function(methods)
     return setmetatable(services, {__index = lookup_helper})
 end
 
-local GCP = {}
-GCP.__index = lookup_helper
+local load_api
+do
+    -- using weak values so that API objects can be garbage-collected
+    local cache = setmetatable({}, { __mode = "v" })
+
+    function load_api(service)
+        if not APIS[service] then
+            error("Unknown API/service: " .. tostring(service), 2)
+        end
+
+        local api = cache[service]
+        if api then
+            return api
+        end
 
-function GCP:new()
-    -- local discoveryUrl = "https://discovery.googleapis.com/discovery/v1/apis"
-    local apis = ApiDiscovery()
-    local servicesInstance = {}
-    for _, service in pairs(apis) do
         local rawAPI = require("resty.gcp.api." .. service)
         local methods = FindApis(rawAPI, {})
-        servicesInstance[service] = BuildMethods(methods)
+        api = BuildMethods(methods)
+
+        cache[service] = api
+        return api
     end
-    local gcp_instance = setmetatable(servicesInstance, GCP)
-    return gcp_instance
+end
+
+local GCP = {}
+GCP.__index = function(self, service)
```

Comment:
maybe `GCP.__newindex`? otherwise we won't hit `self[service]` cache if a api is already loaded 🤔